consensus/istanbul/backend/engine.go

```diff
@@ -411,6 +422,19 @@ func (sb *backend) Prepare(chain consensus.ChainReader, header *types.Header) er
 func (sb *backend) Finalize(chain consensus.ChainReader, header *types.Header, state *state.StateDB, txs []*types.Transaction,
 	receipts []*types.Receipt) (*types.Block, error) {
 
+	// We can assure that if the kip71 hard forked block should have the field of base fee
+	if chain.Config().IsKIP71ForkEnabled(header.Number) {
+		if header.BaseFee == nil {
+			logger.Error("KIP-71 hard forked block should have baseFee", "blockNum", header.Number.Uint64())
+			return nil, errors.New("Invalide KIP-71 block with no baseFee")
+		}
+	} else {
+		if header.BaseFee != nil {
```

Comment:
How about changing this to `else if header.BaseFee != nil` ?
Also how about using `ErrInvalidBaseFee`?

Reply:
1. `else if header.BaseFee != nil` looks better. Thanks
2. @ian0371 suggested me creating the instance when the error occur because It is only used there.

Reply:
I think she meant `errors.New("Invalide block with baseFee")` at line 434 -> `ErrInvalidBaseFee`. My suggestion was about line 429.

Reply:
I confused. It is replaced. Thanks @jimni1222 @ian0371